### DISCUSSION TITLE
Dafoam downloading

### DISCUSSION THREAD

**Discussion by Ryupuriwatt** [mdolab/dafoam#364] [Installation]
Hello professor Ping He, Now I am trying to download Dafoam following the tutorial and now I have got a problem at DAFoam downloading step. Please give me a resolution 🙏🙏🙏🙏

![E2945BFC-8CDC-4476-B911-6B55C54E2371](https://user-images.githubusercontent.com/115110434/194142335-77853670-cb06-4205-ad43-ddbe209e99d1.jpeg)
![7B1025C1-2814-4EA9-AF44-72CB67BC4600](https://user-images.githubusercontent.com/115110434/194142414-4568ba3b-7ff2-4866-9e47-44c680aff200.jpeg)

**Comment by friedenhe** [comment]
It seems that the ADF version of OpenFOAM was not compiled properly. Can you check if you can see dafoam/OpenFOAM/OpenFOAM-v1812-ADF?

**Reply by Ryupuriwatt** [reply]
I can see ADF file. I try to download it many times but always doesn’t work out😭

**Reply by friedenhe** [reply]
Thanks for the feedback. We managed to reproduce the error you saw. It was caused by missing the environment reset line after compiling the ADF version of OpenFOAM. All you previous installations are correct, just run this `unset WM_CODI_AD_MODE &&
. $HOME/dafoam/loadDAFoam.sh` before compiling DAFoam. We have updated the installation documentation to fix this issue.

**Reply by Ryupuriwatt** [reply]
![image](https://user-images.githubusercontent.com/115110434/194742800-c05a0da2-1747-4c41-998d-4d04c073267e.jpg)

Now I have done daFoam regression test but it show me this error could you please give me a resolution?

**Comment by friedenhe** [comment]
Your TACs installation is probably not correct...

**Reply by friedenhe** [reply]
@Ryupuriwatt We have identified the problem. It was caused by the version of the nptyping package. The fix is to do this `pip install nptyping==1.4.4` and then run the test again. We have updated the installation guide to fix this issue. Let me know if it works for you.
